os-autoinst-obs-auto-submit

```diff
@@ -173,37 +194,62 @@ handle_auto_submit() {
     )
 }
 
-prefix="${prefix:-""}"
-[ "$dry_run" = "1" ] && prefix="echo"
-osc="${osc:-"$prefix retry -e -- osc"}"
+get_project_packages() {
+    echo "${packages:-$($osc ls "$dst_project" | grep -v '\-test$')}"
+}
 
-# submit from staging project if specified
-[[ $staging_project && $staging_project != none && $staging_project != "$src_project" ]] && src_project=$staging_project from_staging=1
+has_pending_submission() {
+    # throttle number of submissions to allow only one SR within a certain number of days
+    # note: Avoid using `grep --quiet` here to keep consuming input so osc does not run into "BrokenPipeError: [Errno 32] Broken pipe".
+    if [[ $throttle_days != 0 ]] && $osc request list --project "$dst_project" --type submit --state new,review --mine --days "$throttle_days" | grep 'Created by' > /dev/null; then
+        echo "Skipping submission, there is still a pending SR younger than $throttle_days days."
+        return 1
+    fi
+    return 0
+}
 
-if [[ -e job_post_skip_submission ]]; then
-    echo "Skipping submission, reason: "
-    cat job_post_skip_submission
-    exit 0
-fi
+is_staging_specified() {
+    [[ -n $staging_project && $staging_project != none && $staging_project != "$src_project" ]] && {
+        src_project="$staging_project"
+        echo 1
+    } || echo 0
+}
 
-# throttle number of submissions to allow only one SR within a certain number of days
-# note: Avoid using `grep --quiet` here to keep consuming input so osc does not run into "BrokenPipeError: [Errno 32] Broken pipe".
-if [[ $throttle_days != 0 ]] && $osc request list --project "$dst_project" --type submit --state new,review --mine --days "$throttle_days" | grep 'Created by' > /dev/null; then
-    echo "Skipping submission, there is still a pending SR younger than $throttle_days days."
-    exit 0
-fi
+submit() {
+    # submit from staging project if specified
+    if [[ -e job_post_skip_submission ]]; then
+        echo "Skipping submission, reason: "
+        cat job_post_skip_submission
+        exit 0
+    fi
+    has_pending_submission || exit 0
+    (
+        cd "$TMPDIR"
+        rc=0
+        auto_submit_packages=$(get_project_packages)
+        for package in $auto_submit_packages; do
+            handle_auto_submit "$package" || rc=$?
+        done
+        # delete package from staging project
+        [[ $(is_staging_specified) ]] \
+            && $osc rdelete -m "Cleaning up $package from $staging_project for next submission" "$staging_project" "$package"
+        if [[ ${#failed_packages[@]} -gt 0 ]]; then
+            echo "Failed packages:"
+            for item in "${failed_packages[@]}"; do
+                package="${item%:*}"
+                exit_status="${item#*:}"
```

Comment:
No quotes are required for assignments like these:
```suggestion
                package=${item%:*}
                exit_status=${item#*:}
```
So at least in new code I'd try to avoid needles quotes.